### PR TITLE
NATT-72: Improve the stability of neutron ovs agent check

### PR DESCRIPTION
**Pull request by tonytan4ever**

### CODE REVIEW

**Comment by cloudnull** [comment]
recheck_xenial_newton_deploy

**Comment by cloudnull** [comment]
recheck_bionic_mnaio_osp13_deploy

**Comment by cloudnull** [comment]
recheck_bionic_mnaio_osp13_deploy

File: playbooks/templates/rax-maas/neutron_ovs_agent_check.yaml.j2 [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
+{% set service_host = ansible_nodename + ansible_local['maas']['general']['local_suffix'] | default('') %}
 type        : agent.plugin
 label       : "{{ check_name }}"
 period      : "{{ maas_check_period_override[label] | default(maas_check_period) }}"
 timeout     : "{{ maas_check_timeout_override[label] | default(maas_check_timeout) }}"
 disabled    : "{{ (check_name is match(maas_excluded_checks_regex)) | ternary('true', 'false') }}"
 details     :
     file    : run_plugin_in_venv.sh
-    args    : ["{{ maas_plugin_dir }}/neutron_ovs_agent_check.py"]
+    args    : ["{{ maas_plugin_dir }}/neutron_ovs_agent_check.py", "--protocol", "{{ neutron_client_protocol }}", "--host", "{{ service_host }}", "--fqdn", "{{ ansible_fqdn }}", "{{ internal_vip_address }}"]
```
Comment (npawelek) [inline playbooks/templates/rax-maas/neutron_ovs_agent_check.yaml.j2]: Is there a reason for both `host` and  `fqdn` here? These values are identical, at least on the OSP side. However, I suspect the same on OSA:

`"--host", "overcloud-controller-1.localdomain", "--fqdn", "overcloud-controller-1.localdomain"`

File: playbooks/files/rax-maas/plugins/neutron_ovs_agent_check.py [hunk 2]
```diff
@@ -86,17 +87,68 @@ def check_process_statuses(container_or_host_name, container=None):
...
+    status_ok(m_name='maas_neutron')
+
+    agent_is_up = "Yes"
+    if ovs_agent['admin_state_up'] and not ovs_agent['alive']:
+            agent_is_up = "No"
+
+    if args.host:
+        name = '%s_status' % ovs_agent['binary']
+    elif args.fqdn:
+        name = '%z_status' % ovs_agent['binary']
```
Comment (npawelek) [inline playbooks/files/rax-maas/plugins/neutron_ovs_agent_check.py]: `%z` typo?

The breakout between host and fqdn seem redundant. More on this below.

Comment (tonytan4ever) [reply playbooks/files/rax-maas/plugins/neutron_ovs_agent_check.py]: Followed the same (wrong) code here:
https://github.com/rcbops/rpc-maas/blob/master/playbooks/files/rax-maas/plugins/neutron_service_check.py#L74

Changed it to %s

File: playbooks/templates/rax-maas/neutron_ovs_agent_check.yaml.j2 (hunk 1 above)
Comment (tonytan4ever) [reply playbooks/templates/rax-maas/neutron_ovs_agent_check.yaml.j2]: This is to follow the same pattern as other neutron agent's checks:

https://github.com/rcbops/rpc-maas/blob/master/playbooks/templates/rax-maas/neutron_linuxbridge_agent_check.yaml.j2#L12

**Comment by npawelek** [comment]
recheck_xenial_queens_deploy

**Comment by tonytan4ever** [comment]
recheck_xenial_pike_deploy

**Pull request merged**